client/python/openlineage/client/client.py

```diff
@@ -68,7 +95,7 @@ def __init__(  # noqa: PLR0913
 
         # Make config ellipsis - as a guard value to not try to
         # reload yaml each time config is referred to.
-        self._config: dict[str, dict[str, str]] | None = None
+        self._config: OpenLineageConfig | None = None
```

Comment:
Need to resolve initializing with `from_dict` too - currently, this is used to pass config from Airflow.

```
    @classmethod
    def from_dict(cls: type[_T], config: dict[str, str]) -> _T:
        return cls(transport=get_default_factory().create(config=config))
```

It does not pass other config params than Transport. 

Also, there can be situation where Transport passed from Airflow is different than the one in the otherwise loaded config file, which would cause weird situation:
- used transport is the one which was passed from Airflow
- config field has different transport configuration loaded in memory